### PR TITLE
Cancel placement/purchase

### PR DESCRIPTION
**Pull request by Structed**
Implemented cancellation of tile purchase
Added capability to deselect all Hand/MoneyCards
Enable/Disable Cancel Place Tile Button on the correct occasions Add Tooltip for Cancel Tile placement button
Refactored actually purchasing tiles. Works for both placing a Tile to town and to the Spare Yard Refactored `AbstractCardCollection.remove_card_by_id() to return how many cards were removed Add PurchaseInfo
Add CancelPlaceTileButton + Texture for normal & disabled

### CODE REVIEW

**Comment by Structed** [comment]
Likely broken because of rebase. I guess, use #46 instead

**Comment by Structed** [comment]
Implements #36

File: src/alcaramba/Drawable/Game/Game.gd [hunk 1]
```diff
@@ -45,25 +52,49 @@ func _on_tile_card_selected(card_node: TileCardDrawable):
...
 # if tile is placed add it to town stack and remove selection
 func _on_tile_placed(tile: TileCard):
+	_cancel_place_tile_button.disabled = true
+	
 	# Disconnect button so tile cannot be added to spares after placing them
 	if _spare_tile_add_button.is_connected("pressed", _spare_tiles, "_on_add_to_spares"):
 		_spare_tile_add_button.disconnect("pressed", _spare_tiles, "_on_add_to_spares")
 	
 	if _selected_tile != null:
 		_selected_tile.clear()
```
Comment (zientist) [inline src/alcaramba/Drawable/Game/Game.gd:71]: _selected_tile is not null when placing from spare. Works anyway but removal from spare does not happen in else part

Comment (Structed) [reply src/alcaramba/Drawable/Game/Game.gd:71]: As discussed, let's rework this later

**Comment by zientist** [comment]
test

**Pull request merged**